custom/ewsghana/api.py

```diff
@@ -790,6 +791,80 @@ def sms_user_sync(self, ews_smsuser, **kwargs):
                 sms_user.set_location(couch_location)
         return sms_user
 
+    def _get_total_counts(self, func, limit=1, **kwargs):
+        meta, _ = func(limit=limit, **kwargs)
+        return meta['total_count'] if meta else 0
+
+    def balance_migration(self, date=None):
+        products_count = self._get_total_counts(self.endpoint.get_products)
+        district_count = self._get_total_counts(
+            self.endpoint.get_locations,
+            filters=dict(type='district', is_active=True)
+        )
+        region_count = self._get_total_counts(
+            self.endpoint.get_locations,
+            filters=dict(type='region', is_active=True)
+        )
+        country_count = self._get_total_counts(
+            self.endpoint.get_locations, filters=dict(type='country', is_active=True)
+        )
+        supply_points_count = self._get_total_counts(self.endpoint.get_supply_points, filters=dict(active=True))
+        web_users_count = self._get_total_counts(self.endpoint.get_webusers)
+        sms_users_count = self._get_total_counts(self.endpoint.get_smsusers)
+
+        stats, _ = EWSMigrationStats.objects.get_or_create(domain=self.domain)
+        stats.products_count = products_count
+        stats.locations_count = district_count + region_count + country_count
+        stats.supply_points_count = supply_points_count
+        stats.web_users_count = web_users_count
+        stats.sms_users_count = sms_users_count
+        stats.save()
+
+        for sms_user in iterate_over_api_objects(
+                self.endpoint.get_smsusers, filters=dict(date_updated__gte=date)
```

Comment:
I think it's worth it to do so, since one of the checks is to make sure that every user has a VerifiedNumber, and those can potentially get reassigned on each run